staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured_test.go

```diff
@@ -75,4 +75,34 @@ func TestNilDeletionTimestamp(t *testing.T) {
 	if deletionTimestamp != nil {
 		t.Errorf("unexpected deletion timestamp field: %q", deletionTimestamp)
 	}
+
+	u.SetDeletionTimestamp(nil)
+	clone := u.DeepCopy()
+	if got := clone.GetDeletionTimestamp(); got != nil {
+		t.Errorf("unexpected deletion timestamp in clone: %v, wanted %v", got, nil)
+	}
+}
+
+func TestNilDeletionGracePeriod(t *testing.T) {
+	var u Unstructured
+	del := u.GetDeletionGracePeriodSeconds()
+	if del != nil {
+		t.Errorf("unexpected non-nil deletion grace period: %v", del)
+	}
+	u.SetDeletionGracePeriodSeconds(u.GetDeletionGracePeriodSeconds())
+	del = u.GetDeletionGracePeriodSeconds()
+	if del != nil {
+		t.Errorf("unexpected non-nil deletion grace period: %v", del)
+	}
+	metadata := u.Object["metadata"].(map[string]interface{})
+	deleteGracePeriod := metadata["deletionGracePeriod"]
```

Comment:
These two tests should probably distinguish between `present, nil` and `no value in map` cases and expect something specific.

Reply:
Both are `+optional` and `omitempty` so there should be no value in the map.

Reply:
Am not against this, as a follow-up preferibly.